enhancements/installer/pre-flight-validations.md

```diff
@@ -0,0 +1,194 @@
+---
+title: pre-flight-validations
+authors:
+  - "@mandre"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-05-18
+last-updated: 2020-05-18
+status: implementable
+---
+
+# Pre-flight validations
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+- Will the validations be run automatically on `cluster create` or will it be
+  an explicit action?
+- If explicit action, how will the validations be enabled? E.g. adding
+  a `--dry-run` option to the installer vs. a separate subcommand or even
+  a separate binary.
```

Comment:
Having an explicit command like `--dry-run`  to do the pre-flight check will help IMO. But I am not too opinionated about it as the pre-flight check should be in default code path.